doc/manual/methods.rst

```diff
@@ -265,6 +265,119 @@ Julia its ability to abstractly express high-level algorithms decoupled
 from implementation details, yet generate efficient, specialized code to
 handle each case at run time.
 
+Redefining Methods
+------------------
+
+When redefining a method or adding new methods,
+it is important to realize that these changes don't take effect immediately.
+This is key to Julia's ability to statically infer and compile code to run fast,
+without the usual JIT tricks and overhead.
+Indeed, any new method definition won't be visible to the current runtime environment,
+including Tasks and Threads, or any previously defined ``@generated`` functions.
+Let's start with an example to see what this means::
+
+    julia> function tryeval()
+               @eval newfun() = 1
+               newfun()
+           end
```

Comment:
that's a possible future change (making bindings static in the world environment), but it's not currently enforced (since it isn't directly required to close #265)